classes/utils/Location.py

```diff
@@ -0,0 +1,28 @@
+from .Position import Position
+
+
+class Location(Position):
+
+    def __init__(self, x=0, y=0, z=0, world=None):
+        assert world is not None, 'World should exist !'
+        super(Location, self).__init__(x, y, z)
+        self.world = world
+
+    def clone(self):
+        return Location(self.x, self.y, self.z, self.world)
+
+    def clone_rounded(self):
+        clone = super(Location, self).clone_rounded()
+        clone.world = self.world
+        return clone
+
+    def distance_squared(self, pos):
+        assert self.world == pos.world, 'World must be the same to calculate squared distance between two Location'
```

Comment:
I'd prefer using custom exceptions that way they can be caught and logged to console.

Reply:
Done, changed to ValueError

Reply:
To test code, you can launch it with --test parameter (`python start.py --test`) or via `pytest test` while beeing at the root of the project